modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java

```diff
@@ -0,0 +1,364 @@
+
+        distributionZoneManager.createZone(new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).build()).get();
+
+        distributionZoneManager.alterZone(
+                ZONE_NAME,
+                new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).dataNodesAutoAdjust(100).build()
+        ).get();
+
+        distributionZoneManager.alterZone(
+                ZONE_NAME,
+                new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).dataNodesAutoAdjust(1000).build()
+        ).get();
+
+        assertZonesChangeTriggerKey(3);
+    }
+
+    @Test
+    void testDataNodesNotPropagatedAfterZoneCreation() throws Exception {
+        Set<ClusterNode> clusterNodes = Set.of(new ClusterNode("1", "name1", null));
+
+        mockCmgLocalNodes(clusterNodes);
+
+        keyValueStorage.put(zonesChangeTriggerKey().bytes(), ByteUtils.longToBytes(100));
+
+        distributionZoneManager.createZone(new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).build()).get();
+
+        verify(keyValueStorage, timeout(1000).times(1)).invoke(any());
+
+        assertZonesChangeTriggerKey(100);
+
+        assertDataNodesForZone(1, null);
+    }
+
+    @Test
+    void testDataNodesNotPropagatedAfterZoneUpdate() throws Exception {
+        Set<ClusterNode> clusterNodes = Set.of(new ClusterNode("1", "name1", null));
+
+        LogicalTopologySnapshot logicalTopologySnapshot = mockCmgLocalNodes(clusterNodes);
+
+        distributionZoneManager.createZone(new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).build()).get();
+
+        assertDataNodesForZone(1, clusterNodes);
+
+        var clusterNodes2 = Set.of(
+                new ClusterNode("1", "name1", null),
+                new ClusterNode("2", "name2", null)
+        );
+
+        when(logicalTopologySnapshot.nodes()).thenReturn(clusterNodes2);
+
+        keyValueStorage.put(zonesChangeTriggerKey().bytes(), ByteUtils.longToBytes(100));
+
+        distributionZoneManager.alterZone(
+                ZONE_NAME,
+                new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).dataNodesAutoAdjust(100).build()
+        ).get();
+
+        verify(keyValueStorage, timeout(1000).times(2)).invoke(any());
+
+        assertZonesChangeTriggerKey(100);
+
+        assertDataNodesForZone(1, clusterNodes);
+    }
+
+    @Test
+    void testZoneDeleteDoNotRemoveMetaStorageKey() throws Exception {
+        Set<ClusterNode> clusterNodes = Set.of(new ClusterNode("1", "name1", null));
+
+        mockCmgLocalNodes(clusterNodes);
+
+        distributionZoneManager.createZone(new DistributionZoneConfigurationParameters.Builder(ZONE_NAME).build());
+
+        assertDataNodesForZone(1, clusterNodes);
+
+        keyValueStorage.put(zonesChangeTriggerKey().bytes(), ByteUtils.longToBytes(100));
+
+        distributionZoneManager.dropZone(ZONE_NAME);
+
+        verify(keyValueStorage, timeout(1000).times(2)).invoke(any());
+    }
```

Comment:
Where is the check that meta storage key has not removed?

Reply:
thanks, fixed